pkg/storage/tpr/install_types.go

```diff
@@ -34,31 +37,71 @@ var thirdPartyResources = []v1beta1.ThirdPartyResource{
 	serviceBindingTPR,
 }
 
+//Installer takes in a client and exposes method for installing third party resources
+type Installer struct {
+	tprs extensionsv1beta.ThirdPartyResourceInterface
+}
+
+//NewInstaller is used to install third party resources
+func NewInstaller(tprs extensionsv1beta.ThirdPartyResourceInterface) *Installer {
+	return &Installer{
+		tprs: tprs,
+	}
+}
+
 // InstallTypes installs all third party resource types to the cluster
-func InstallTypes(cs clientset.Interface) error {
-	tprs := cs.Extensions().ThirdPartyResources()
+func (i *Installer) InstallTypes() error {
+	var wg sync.WaitGroup
+	errMsg := make(chan string, len(thirdPartyResources))
+
 	for _, tpr := range thirdPartyResources {
 		glog.Infof("Checking for existence of %s", tpr.Name)
-		if _, err := tprs.Get(tpr.Name); err == nil {
+		if _, err := i.tprs.Get(tpr.Name); err == nil {
 			glog.Infof("Found existing TPR %s", tpr.Name)
 			continue
 		}
 
 		glog.Infof("Creating Third Party Resource Type: %s", tpr.Name)
-		if _, err := tprs.Create(&tpr); err != nil {
-			glog.Errorf("Failed to create Third Party Resource Type: %s (%s))", tpr.Name, err)
-			return err
-		}
-		glog.Infof("Created TPR '%s'", tpr.Name)
-		// There can be a delay, so poll until it's ready to go...
-		for i := 0; i < 30; i++ {
-			if _, err := tprs.Get(tpr.Name); err != nil {
-				glog.Infof("TPR %s is ready", tpr.Name)
-				break
+
+		wg.Add(1)
+		go func(tpr v1beta1.ThirdPartyResource, tprInfce extensionsv1beta.ThirdPartyResourceInterface) {
```

Comment:
Bit of a nit too, but tprInfce is a wonky name. I'd expect tprIfc or tprI or something. But if others are fine with this, I'm fine :)